samcli/hook_packages/terraform/hooks/prepare/hook.py

```diff
@@ -167,3 +141,56 @@ def _update_resources_paths(cfn_resources: Dict[str, Any], terraform_application
                 original_path = resource.get("Properties", {}).get(attribute)
                 if isinstance(original_path, str) and not os.path.isabs(original_path):
                     resource["Properties"][attribute] = str(Path(terraform_application_dir).joinpath(original_path))
+
+
+def _generate_plan_file(skip_prepare_infra: bool, terraform_application_dir: str) -> dict:
```

Comment:
should we move the try/except block as well to this function, as all the errors should be coming from executing Terraform commands

Reply:
I moved the `CalledProcessError` and `LoadingPatternError` exceptions to the function but left the `OSError` in the parent function since `open()` will throw this if it fails.

Reply:
thanks Daniel :)